### DISCUSSION TITLE
Possible bug? Versions after 3.2.5 UI don't update knob values.

### DISCUSSION THREAD

**Discussion by Chris-Spy** [michaelwillis/dragonfly-reverb#160] [General]
Hi Michael,

Just letting you know for a possible bug. All versions after 3.2.5 (VST, VST3 & CLAP) seem to have a problem updating the knob values when I turn them. The values update if I turn off the UI (in Reaper) change the values with the scrollbars and then turn on the UI again.

Thanks!

**Comment by michaelwillis** [comment]
Hi @Chris-Spy, today I tested the 3.2.10 VST3 and CLAP plugins with REAPER on MacOS and Linux. I don't experience the bug that you describe. What operating system are you using?

This is how I tried to reproduce the bug:
* Create a track with a piano virtual instrument
* Add one of the Dragonfly Reverb plugins as FX after the piano
* Hit a key on the virtual piano, verify that it sounds as expected
* Dial up the decay time to maximum
* Hit a key on the virtual piano, verify that the reverb rings out for a long time.
* Repeat with other dials that have obvious effects, like diffuse, modulation, high cut.

Am I missing something?

P.S. Are you [Chris Spyratos](http://scoringcentral.mattiaswestlund.net/member.php?action=profile&uid=72)? I miss when that forum had more activity.

**Comment by Chris-Spy** [comment]
Hi @michaelwillis ! Yes, I am that guy :)

Totally forgot to mention I am running Reaper in Windows 10. The plugin sounds as expected, only the UI numbers don't respond to the changes. I made a capture to illustrate it. Notice also how the Dry/Early/Early Send & Late faders update their numbers as they should. They even force the whole UI to update, restoring every knob number to their assigned values.

![dragonfly-knob-values](https://github.com/michaelwillis/dragonfly-reverb/assets/135475579/620b7a8f-549e-437f-9789-6847bd350918)

**Comment by michaelwillis** [comment]
Thanks for the clarification, that's definitely a bug. I can't promise a fix, since I don't have a machine with Windows 10 (or any other version of Windows). It's a good clue to see that the sliders properly update the whole UI, so I can review the difference between the code that handles the sliders and the dials. 

Do you have another DAW or plugin host that you can use to test this? I'm curious if it's specifically Reaper or if you'll see the same bug running the plugins in a different host.

**Reply by Chris-Spy** [reply]
I get the same results inside VSTHost as well as inside bespoke synth. If I can find another Windows PC to test it I will keep you updated!

**Comment by falkTX** [comment]
This seems similar to https://github.com/AidaDSP/AIDA-X/issues/20 where the UI is not fully repainted properly leading to old stuff shown on screen.
I was able to reproduce this myself, suspect it to be some pugl related thing. A workaround is to find a way to refresh/redraw the UI, how to do that depends on the host.

**Comment by davenose** [comment]
I don't experience this with my current configuration (Windows 10 22H2, Cubase 12.0.60, Reaper 6.81), with either the VST/VST3 versions of Dragonfly 3.2.10 in either DAW.

However,  I *have* previously experienced the lack of knob value updates with an earlier version of Reaper (6.7x) and the VST3 Dragonfly 3.2.10 versions.

**Reply by davenose** [reply]
I also just tested with VST Host 1.57.  I do not experience the lack of knob updates on any of Dragonfly reverbs with VST Host (both VST, VST3 variants for all plugins - Early Reflections/Plate/Room/Hall).

@Chris-Spy - with what VST Host version were you able to reproduce the behavior?
